backend/app/services/adapters/executor_kinds.py

```
@@ -603,6 +607,70 @@ async def delete_executors_async():
+                if subtask.role == SubtaskRole.USER:
+                    user_message = subtask.prompt
+                    task_start_time = subtask.created_at.strftime('%Y-%m-%d %H:%M:%S') if isinstance(subtask.created_at, datetime) else subtask.created_at
+                if subtask.role == SubtaskRole.ASSISTANT:
+                    task_end_time = subtask.updated_at.strftime('%Y-%m-%d %H:%M:%S') if isinstance(subtask.updated_at, datetime) else subtask.updated_at
+
+            user_name = "Unknown"
+            if user_id:
+                user = db.query(User).filter(User.id == user_id).first()
+                user_name = user.user_name
+
+            task_type = task_crd.metadata.labels and task_crd.metadata.labels.get("taskType") or "chat"
```

Ruff (0.14.5)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)